R/Google.R

```diff
@@ -0,0 +1,207 @@
+          by = c("key", "date")
+        ) %>%
+        rename(
+          level_1_region_code = .data$`3166-1-alpha-3`,
+          level_1_region = .data$country_name,
+          level_2_region_code = .data$subregion1_code,
+          level_2_region = .data$subregion1_name,
+          level_3_region_code = .data$subregion2_code,
+          level_3_region = .data$subregion2_name,
+          cases_new = .data$new_confirmed,
+          cases_total = .data$total_confirmed,
+          deaths_new = .data$new_deceased,
+          deaths_total = .data$total_deceased,
+          tested_new = .data$new_tested,
+          tested_total = .data$total_tested,
+          recovered_new = .data$new_recovered,
+          recovered_total = .data$total_recovered,
+          hosp_new = .data$new_hospitalized,
+          hosp_total = .data$total_hospitalized
+        ) %>%
+        mutate(
+          date = ymd(.data$date),
+          cases_new = as.numeric(.data$cases_new),
+          cases_total = as.numeric(.data$cases_total),
+          deaths_new = as.numeric(.data$deaths_new),
+          deaths_total = as.numeric(.data$deaths_total),
+          recovered_new = as.numeric(.data$recovered_new),
+          recovered_total = as.numeric(.data$recovered_total),
+          tested_new = as.numeric(.data$tested_new),
+          tested_total = as.numeric(.data$tested_total),
+          hosp_new = as.numeric(.data$hosp_new),
+          hosp_total = as.numeric(.data$hosp_total)
+        ) %>%
+        replace_na(
+          list(
+            level_1_region = "Unknown",
+            level_2_region = "Unknown"
+          )
+        ) %>%
+        select(
+          date,
+          level_1_region,
+          level_1_region_code,
+          level_2_region,
+          level_2_region_code,
+          cases_new,
+          cases_total,
+          deaths_new,
+          deaths_total,
+          recovered_new,
+          recovered_total,
+          hosp_new,
+          hosp_total,
+          tested_new,
+          tested_total,
+          everything()
+        )
+    },
+
+    #' @description Google specific subregion level data cleaning. Takes the
+    #' data cleaned by `clean_common` and aggregates it to the country level
+    #' (level 1).
+    #' @importFrom dplyr select summarise group_by across everything
+    #' @importFrom rlang .data
+    clean_level_1 = function() {
+      self$data$clean <- self$data$clean %>%
+        group_by(
+          .data$date, .data$level_1_region_code,
+          .data$level_1_region
+        ) %>%
+        summarise(across(where(is.double), sum))
+    },
+
+    #' @description Google specific subregion2 level data cleaning. Takes the
+    #' data cleaned by `clean_common` and aggregates it to the subregion level
+    #' (level 2).
+    #' @importFrom dplyr select summarise group_by across everything
+    #' @importFrom rlang .data
+    clean_level_2 = function() {
+      self$data$clean <- self$data$clean %>%
```

Comment:
what controls here for places that have national data but no level 2 data? Similarly for level 3 data?